sql/src/main/java/io/crate/planner/optimizer/Optimizer.java

```diff
@@ -41,22 +42,25 @@
 
     private final List<Rule<?>> rules;
     private final Supplier<Version> minNodeVersionInCluster;
+    private final Functions functions;
 
-    public Optimizer(List<Rule<?>> rules, Supplier<Version> minNodeVersionInCluster) {
-        this.rules = rules;
+    public Optimizer(Functions functions, Supplier<Version> minNodeVersionInCluster, List<Class<? extends Rule<?>>> rulesToInclude) {
+        this.rules = loadRules(rulesToInclude);
```

Comment:
```suggestion
    this.rules = new LoadedRules().getRules(rulesToInclude);
```

`loadRules` is a shallow method. I don't think there is a benefit in having a separate method?